xchange-bitmex/src/main/java/org/knowm/xchange/bitmex/BitmexAdapters.java

```diff
@@ -352,37 +372,58 @@ private static OrderType convertType(String side) {
   public static FundingRecord adaptFundingRecord(BitmexWalletTransaction walletTransaction) {
 
     String datePattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
-
     SimpleDateFormat dateFormat = new SimpleDateFormat(datePattern);
-    Date dateFunding = null;
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
 
-    try {
-      dateFunding = dateFormat.parse(walletTransaction.getTransactTime());
-    } catch (ParseException e) {
-      throw new IllegalArgumentException(e);
+    Date dateFunding = null;
+    if (walletTransaction.getTransactTime() != null) {
+      try {
+        dateFunding = dateFormat.parse(walletTransaction.getTransactTime());
```

Comment:
we could just use `Date` instead of `String` for the `WalletTransaction#transactTime` attribute and proper Jackson annotations to make Jackson parse this

Reply:
Done, see 8014b9504e8811a175bed5ec506bd54f878ccb27